Taskfile.yml

```diff
@@ -63,21 +63,56 @@ tasks:
     desc: Run the unit tests.
     cmds:
       # One day I will understand how to use -coverpkg=./... :-(
-      - go test -count=1 -short -coverprofile=coverage.out ./...
+      # FIXME - go test -count=1 -short -coverprofile=coverage.out ./...
+      - go test -count=1 -short -coverprofile=coverage.out ./cogito ./cmd/cogito
 
   test:all:
     desc: Run all the tests (unit + integration). Use this target to get total coverage.
     cmds:
-      - go test -count=1 -coverprofile=coverage.out ./...
+      # FIXME - go test -count=1 -coverprofile=coverage.out ./...
+      - go test -count=1 -short -coverprofile=coverage.out ./cogito ./cmd/cogito
     env: *test-env
 
   test:smoke:
     desc: Simple smoke test of the local executables.
     cmds:
+      - task: build
+      - task: build:copydir
+      - task: test:buildinfo
       - echo '{{.SMOKE_INPUT}}' | ./bin/check
       - echo
       - echo '{{.SMOKE_INPUT}}' | ./bin/in dummy-dir
       - echo
+      - rm -rf /tmp/cogito-test
+      - mkdir -p /tmp/cogito-test
+      - >
+        ./bin/copydir cogito/testdata/one-repo/a-repo /tmp/cogito-test --dot 
+        --template repo_url=https://github.com/foo/bar head=dummyHead
+        branch_name=dummyBranch commit_sha=dummySha
+      - echo '{{.PUT_INPUT}}' | ./bin/out /tmp/cogito-test
```

Comment:
I hope I will be able to keep this smoke test of the put step, but it will need to interact with GitHub...